### DISCUSSION TITLE
Would you use unofficial response status codes?

### DISCUSSION THREAD

**Discussion by jansupol** [eclipse-ee4j/jersey#5441] [Polls]
Jakarta REST `Response.Status` enum contains just a selected list of `Response.StatusTypes`.

While it is fine to have a list of all official Status codes maintained by [the IANA](https://www.iana.org/assignments/http-status-codes/http-status-codes.xhtml), there is also a list of status codes that are not officially supported by an RFC, but used by the vendors, such as Apache, AWS, Spring, or Ngnix (see for instance [wiki](https://en.wikipedia.org/wiki/List_of_HTTP_status_codes)).

**Comment by gluser1357** [comment]
In this context, it would be nice to have a Response.notFound() shortcut to substitute Response.status(Status.NOT_FOUND).

**Reply by jansupol** [reply]
Not found is usually generated by the framework when the resource is not found. Do you generate 404 on your own that often?

**Reply by gluser1357** [reply]
Yes, sometimes, if a resource at /path/to/{id} doesn't exist in the database.

**Reply by mkarg** [reply]
Why not simply return `null` or throw `NotFoundException` in that case and handle the response code outside of the resource?
